alternator/rjson.cc

```diff
@@ -113,6 +113,48 @@ void push_back(rjson::value& base_array, rjson::value&& item) {
 
 }
 
+bool single_value_comp::operator()(const rjson::value& r1, const rjson::value& r2) const {
+       auto r1_type = r1.GetType();
+       auto r2_type = r2.GetType();
+       switch (r1_type) {
+       case rjson::type::kNullType:
+           return r1_type < r2_type;
+       case rjson::type::kFalseType:
+           return r1_type < r2_type;
+       case rjson::type::kTrueType:
+           return r1_type < r2_type;
+       case rjson::type::kObjectType:
+           throw rjson::error("Object type comparison is not supported");
```

Comment:
So, if I r1 is True and r2 is an object, the result is 1, but if it's the opposite order - r1 is object and r2 is true, the result is an exception? This doesn't make any sense....
Also, throwing an exception like this will cause - unless caught - the user to experience and "internal error" instead of a normal API error.

Reply:
right... something is off here, unfortunately this code is no longer fresh in my mind at all. I'll take a look at it all and perhaps fix these issues as a separate patch, unrelated to this series.

Reply:
Ok, makes sense. So I'll commit your next version without insisting you fix these pre-existing problems now (although, maybe we should have an issue about it to remember it? Or something?)

Reply:
I vaguely remember that this type comparison was added as a "smart" way of using the fact that `kNullType < kFalseType < kTrueType`... well, it doesn't look that smart any more once you consider comparing `kTrueType` against `kObjectType`...